app/models/storage_location.rb

```diff
@@ -81,6 +81,12 @@ def size
     inventory_items.sum(:quantity)
   end
 
+  def total_active_inventory_count
+    inventory_items.select{ |item| 
+      Item.find(item.item_id).active 
+    }.sum(&:quantity)
+  end
```

Comment:
Ditto here - 

```ruby
self.inventory_items.
includes(:item).
where(items: { active: true}).
select('items.quantity').
sum(:quantity)
```

This will do the sum on the quantity column itself and rely entirely on the database rather than in-memory operations.

There is some overlap between this method and the previous one, so it might make sense to add a method on `storage_location` to get the active items belonging to it.